### PR TITLE
CLOUDP-127414: PE cleanup action

### PR DESCRIPTION
**Pull request by Sugar-pack**
### All Submissions:

Add a new step to Clean Atlas Organization action. It is needed when e2e tests are stopped manually or panicked. In this case, there may be unremoved private endpoints.

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: .github/actions/cleanup-pe/cleantagged.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"google.golang.org/api/compute/v1"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/actions/cloud"
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/api/gcp"
+)
+
+func cleanAllTaggedAWSPE(region, tagName, tagValue string) error {
```
Comment (fabritsius) [inline .github/actions/cleanup-pe/cleantagged.go:20]: All of the "tagged" function versions look a lot like their not-tagged counterparts (close to duplication). What do you think about using the same cleaning function for tagged and non-tagged? You can add a flag or just make a decision based on the fact that tagName and tagValue are both empty.

File: .github/actions/cleanup-pe/cleanPE.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	switch cleanAllEnv {
+	case "true":
+		cleanOnlyTaggedPE = true
+	case "false":
+	default:
+		log.Fatal("CLEAN_TAGGED_PE must be set to true or false")
+	}
+	err := SetGCPCredentials()
+	if err != nil {
+		log.Fatal(err)
```
Comment (Julien-Ben) [inline .github/actions/cleanup-pe/cleanPE.go:26]: Should we interrupt execution after a fatal error occurs ?

File: .github/actions/cleanup-pe/cleantagged.go (hunk 1 above)
Comment (fabritsius) [reply .github/actions/cleanup-pe/cleantagged.go:20]: Or make 2 wrapper functions on top of the same core function might be also a great idea

File: .github/actions/cleanup-pe/cleanPE.go (hunk 2 above)
Comment (Sugar-pack) [reply .github/actions/cleanup-pe/cleanPE.go:26]: I think so. The program will stop if it doesn't get the credentials or if it receives an error from another function. Log.Fatal is equivalent to Print() followed by a call to os.Exit(1).

Comment (Julien-Ben) [reply .github/actions/cleanup-pe/cleanPE.go:26]: Oh I wasn't aware that the logging library was taking care of interrupting the execution, all good then !

**Review by fabritsius — APPROVED** [review]
Looks Great 👍

**Review by fabritsius — APPROVED** [review]

**Pull request merged**